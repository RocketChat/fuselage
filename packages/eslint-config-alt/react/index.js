module.exports = {
  extends: '../minimal',
  plugins: ['react', 'react-hooks'],
  parser: '@babel/eslint-parser',
  rules: {
    'jsx-quotes': ['error', 'prefer-single'],
    'react/display-name': 'error',
    'react/jsx-uses-react': 'error',
    'react/jsx-uses-vars': 'error',
    'react/jsx-no-undef': 'error',
    'react/jsx-fragments': ['error', 'syntax'],
    'react/no-multi-comp': 'error',
    'react/react-in-jsx-scope': 'error',
    'react-hooks/rules-of-hooks': 'error',
    'react-hooks/exhaustive-deps': 'warn',
  },
  settings: {
    'import/resolver': {
      node: {
        extensions: ['.js', '.jsx', '.ts', '.tsx'],
      },
    },
    'react': {
      version: 'detect',
    },
  },
  env: {
    browser: true,
    es6: true,
  },
  overrides: [
    {
      files: ['**/*.ts', '**/*.tsx'],
      extends: '../typescript',
      parser: '@typescript-eslint/parser',
      plugins: ['react', 'react-hooks'],
      rules: {
        'jsx-quotes': ['error', 'prefer-single'],
        'react/display-name': 'error',
        'react/jsx-uses-react': 'error',
        'react/jsx-uses-vars': 'error',
        'react/jsx-no-undef': 'error',
        'react/jsx-fragments': ['error', 'syntax'],
        'react/no-multi-comp': 'error',
        'react/react-in-jsx-scope': 'error',
        'react-hooks/rules-of-hooks': 'error',
        'react-hooks/exhaustive-deps': 'warn',
      },
      env: {
        browser: true,
        es6: true,
      },
      settings: {
        'import/resolver': {
          node: {
            extensions: ['.js', '.jsx', '.ts', '.tsx'],
          },
        },
        'react': {
          version: 'detect',
        },
      },
    },
    {
      files: [
        '**/*.stories.js',
        '**/*.stories.jsx',
        '**/*.stories.ts',
        '**/*.stories.tsx',
      ],
      rules: {
        'react/display-name': 'off',
        'react/no-multi-comp': 'off',
      },
    },
  ],
};
