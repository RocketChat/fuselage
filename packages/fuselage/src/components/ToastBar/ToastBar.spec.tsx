import { render } from '@testing-library/react';
import { axe } from 'jest-axe';
import React from 'react';

import { ToastBar } from '.';

describe('[ToastBar Component]', () => {
  it('renders without crashing', () => {
    render(<ToastBar />);
  });

  it('should have no a11y violations', async () => {
    const { container } = render(<ToastBar />);

    const results = await axe(container);
    expect(results).toHaveNoViolations();
  });
});
