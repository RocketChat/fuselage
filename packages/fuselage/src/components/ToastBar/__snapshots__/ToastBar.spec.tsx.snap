// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[ToastBar Rendering] renders Default without crashing 1`] = `
<body>
  <div>
    <div
      class="rcx-box rcx-box--full rcx-css-fnyeze rcx-toastbar rcx-toastbar--info  rcx-css-n4l2cd"
    >
      <div
        class="rcx-toastbar_inner"
      >
        <i
          aria-hidden="true"
          class="rcx-box rcx-box--full rcx-icon--name-info rcx-icon  rcx-toastbar_icon--info rcx-css-4pvxx3"
        >
          
        </i>
        <div
          class="rcx-toastbar_content"
          role="alert"
        >
          Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod
  tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam,
  quis nostrud exercitation ullamco
        </div>
      </div>
      <div
        class="rcx-box rcx-box--full rcx-css-1rc3wl5 rcx-toastbar_progressbar"
      />
    </div>
  </div>
</body>
`;

exports[`[ToastBar Rendering] renders Error without crashing 1`] = `
<body>
  <div>
    <div
      class="rcx-box rcx-box--full rcx-css-fnyeze rcx-toastbar rcx-toastbar--error  rcx-css-n4l2cd"
    >
      <div
        class="rcx-toastbar_inner"
      >
        <i
          aria-hidden="true"
          class="rcx-box rcx-box--full rcx-icon--name-ban rcx-icon  rcx-toastbar_icon--error rcx-css-4pvxx3"
        >
          
        </i>
        <div
          class="rcx-toastbar_content"
          role="alert"
        >
          Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod
  tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam,
  quis nostrud exercitation ullamco
        </div>
      </div>
      <div
        class="rcx-box rcx-box--full rcx-css-1rc3wl5 rcx-toastbar_progressbar"
      />
    </div>
  </div>
</body>
`;

exports[`[ToastBar Rendering] renders Success without crashing 1`] = `
<body>
  <div>
    <div
      class="rcx-box rcx-box--full rcx-css-fnyeze rcx-toastbar rcx-toastbar--success  rcx-css-n4l2cd"
    >
      <div
        class="rcx-toastbar_inner"
      >
        <i
          aria-hidden="true"
          class="rcx-box rcx-box--full rcx-icon--name-circle-check rcx-icon  rcx-toastbar_icon--success rcx-css-4pvxx3"
        >
          
        </i>
        <div
          class="rcx-toastbar_content"
          role="alert"
        >
          Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod
  tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam,
  quis nostrud exercitation ullamco
        </div>
      </div>
      <div
        class="rcx-box rcx-box--full rcx-css-1rc3wl5 rcx-toastbar_progressbar"
      />
    </div>
  </div>
</body>
`;

exports[`[ToastBar Rendering] renders TinyText without crashing 1`] = `
<body>
  <div>
    <div
      class="rcx-box rcx-box--full rcx-css-fnyeze rcx-toastbar rcx-toastbar--info  rcx-css-n4l2cd"
    >
      <div
        class="rcx-toastbar_inner"
      >
        <i
          aria-hidden="true"
          class="rcx-box rcx-box--full rcx-icon--name-info rcx-icon  rcx-toastbar_icon--info rcx-css-4pvxx3"
        >
          
        </i>
        <div
          class="rcx-toastbar_content"
          role="alert"
        >
          Lorem ipsum dolor sit amet
        </div>
      </div>
      <div
        class="rcx-box rcx-box--full rcx-css-1rc3wl5 rcx-toastbar_progressbar"
      />
    </div>
  </div>
</body>
`;

exports[`[ToastBar Rendering] renders WithCloseButton without crashing 1`] = `
<body>
  <div>
    <div
      class="rcx-box rcx-box--full rcx-css-fnyeze rcx-toastbar rcx-toastbar--info  rcx-css-n4l2cd"
    >
      <div
        class="rcx-toastbar_inner"
      >
        <i
          aria-hidden="true"
          class="rcx-box rcx-box--full rcx-icon--name-info rcx-icon  rcx-toastbar_icon--info rcx-css-4pvxx3"
        >
          
        </i>
        <div
          class="rcx-toastbar_content"
          role="alert"
        >
          Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod
  tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam,
  quis nostrud exercitation ullamco
        </div>
        <div
          class="rcx-toastbar-close"
        >
          <button
            aria-label="Dismiss alert"
            class="rcx-box rcx-box--full rcx-button--tiny-square rcx-button--square rcx-button--icon rcx-button"
            type="button"
          >
            <i
              aria-hidden="true"
              class="rcx-box rcx-box--full rcx-icon--name-cross rcx-icon rcx-css-1g87xs3"
            >
              
            </i>
          </button>
        </div>
      </div>
      <div
        class="rcx-box rcx-box--full rcx-css-1rc3wl5 rcx-toastbar_progressbar"
      />
    </div>
  </div>
</body>
`;
